packages/components/src/FormField/hooks/useFormFieldFocus.ts

```diff
@@ -8,27 +8,43 @@ interface UseFormFieldFocusProps {
   wrapperRef?: React.RefObject<HTMLDivElement>;
 }
 
+const FOCUS_DATA_ATTRIBUTE = "data-atl-maintain-portal-focus";
+
+export const focusAttribute = {
+  [FOCUS_DATA_ATTRIBUTE]: "true",
+};
+
 export function useFormFieldFocus(
   props: UseFormFieldFocusProps,
 ): UseFormFieldFocus {
   const [focused, setFocused] = useState(false);
+  const activeElement = useActiveElement();
+  const activeElementRef = useRef(activeElement);
+
+  useEffect(() => {
+    activeElementRef.current = activeElement;
+  }, [activeElement]);
+
+  function handleFocusIn() {
+    setFocused(true);
+  }
+
+  function handleFocusOut() {
+    setTimeout(() => {
+      const focusedElementWithinWrapper = props.wrapperRef?.current?.contains(
+        document.activeElement,
+      );
+      const focusException = activeElementRef.current?.closest(
+        `[${FOCUS_DATA_ATTRIBUTE}='true']`,
+      );
```

Comment:
this works but it's a little weird

the problem I'm solving is that the toolbar hides when the `toolbarVisibility` is `whileEditing`. the way it hides is it checks if the descendants of the toolbar are focused or not, and then sets some state depending on that

now that the Menu list is being portalled, it is no longer a descendant. it's just some element on the `body`, and so we need some way to disambiguate that element on the body from any other element on the body

then we can whitelist it and say please don't lose focus if the current active element has a parent somewhere up the tree that has `"data-atl-maintain-portal-focus"`

we will have to be careful about using the attribute. it must only be applied to somethign that is portalled to the body and can never be a parent of a very broad tree - otherwise if we give focus to some other element that is a descendant that should cause the toolbar to lose focus, it would  not